### DISCUSSION TITLE
Docker images caching on the camera

### DISCUSSION THREAD

**Discussion by igor-rud-kr** [AxisCommunications/acap-computer-vision-sdk-examples#66] [Q&A] [answered]
When testing https://github.com/AxisCommunications/acap-computer-vision-sdk-examples/tree/main/object-detector-python example, I noticed that model docker image is cached somewhere in the camera. When I change Dockerfile.model and config/env.aarch64.artpec8 , rebuild and load the image to the camera (please see commands below), it has no effect. 

# Build and upload inference models
docker build . -f Dockerfile.model -t $MODEL_NAME --build-arg ARCH
docker save $MODEL_NAME | docker --tlsverify -H tcp://$AXIS_TARGET_IP:$DOCKER_PORT load

The following command
docker --tlsverify --tlscacert=ca.pem --tlscert=cert.pem --tlskey=key.pem -H tcp://$AXIS_TARGET_IP:$DOCKER_PORT system prune -af
does not help either

Restarting Docker from camera web interface does not help. Removing and installing Docker from camera web interface does not help does not help either.  

The only reliable way to make the camera see new image is to stop and remove docker, log in to the camera, remove physically docker files from the storage and re-install Docker again. This is very inconvenient

**Comment by Corallo** [comment, marked as answer]
Hi Igor

You have to delete the old docker volumes from the camera.
The clean way is to `docker-compose --tlsverify -H tcp://$AXIS_TARGET_IP:$DOCKER_PORT --env-file ./config/env.$ARCH.$CHIP down -v` like explained in the example, after stopping the container.

**Reply by igor-rud-kr** [reply]
Yes, this did work. I did not realize that pressing ctrl-C is not equivalent to the command above. Thank you!
